packages/amplify-graphql-transformer-core/src/transformer-context/resource-helper.ts

```diff
@@ -1,20 +1,41 @@
-import { GraphQLAPIProvider, TransformerResourceHelperProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { GraphQLAPIProvider, TransformerResourceHelperProvider, ModelFieldMap } from '@aws-amplify/graphql-transformer-interfaces';
 import { CfnParameter, Token } from '@aws-cdk/core';
 import { StackManager } from './stack-manager';
 import md5 from 'md5';
+import { ModelResourceIDs } from 'graphql-transformer-common';
+import { ModelFieldMapImpl } from './model-field-map';
 
+/**
+ * Contains helper methods for transformers to access and compile context about resource generation
+ */
 export class TransformerResourceHelper implements TransformerResourceHelperProvider {
   private api?: GraphQLAPIProvider;
-  // eslint-disable-next-line no-useless-constructor
-  constructor(private stackManager: StackManager) {}
-  public generateResourceName = (name: string): string => {
+
+  // a mapping of models that have been renamed with @mapsTo
+  readonly #modelNameMap = new Map<string, string>();
```

Comment:
I am not sure if this logic should be here in the context. The idea with transformer v2 was the DataSourceProvider would be in charge of naming the tables and resources, and enhancer would go and modify thing not in global state but using the DataSourceProvider instance and call some methods on it.

Reply:
If you have other suggestions for where this should go, I'm willing to entertain ideas. Are there any examples of how a DataSourceEnhancer works?

But one thing is there needs to be some information about renames as the AST is being walked. Both so that the @mapsTo transformer can register renames as it discovers them and also so that relational resolvers can register renamed foreignKey fields as it finds them. It also seems like it would be easier to generate the table and resolver names correctly the first time rather than have a step at the end to "correct" everything.

Reply:
We dont have any examples of DataSourceEnhancer interacting with DataSourceProvider. The bases class might be missing some methods needed to update the table names. When porting we probably followed the same pattern as V1.

One way I can think of this to work is have a method in the ModelTransformer exposing a method which can set the table name and the `mapsTo` transformer using that. I am not sure if this is too late for that though.